src/modules/exchange/exchangeApi.js

```diff
@@ -43,3 +45,22 @@ function makeExchangeCacheApi (ai: ApiInput) {
 
   return out
 }
+
+export type ExchangeSwapRate = {
+  pair: string,
+  rate: string,
+}
+
+const API_PREFIX = 'https://shapeshift.io'
+
+export function makeShapeshiftApi (io: FixedIo) {
+  return {
+    async getExchangeSwapRate (fromCurrency: string, toCurrency: string) {
+      const pair = `${fromCurrency}_${toCurrency}`
+      const reply = await io.fetch(`${API_PREFIX}/rate/${pair}`)
+      const json: ExchangeSwapRate = await reply.json()
```

Comment:
When I tested this with a fake server, I got back a 404 error like I expected. However, the 404 included valid JSON as its message body. Therefore, this code returned the error JSON as if it were the exchange data. We eventually need some `if (!reply.ok) { throw new Error(`The server returned HTTP error code ${reply.status}`) }` logic.